src/components/CheckboxInput/CheckboxInput.tsx

```diff
@@ -25,47 +24,22 @@ export type CheckboxInputProps = CheckboxHTMLElementProps & {
   id: string;
 };
 
-const checkboxIconNameMap = {
-  indeterminate: 'checkbox-indeterminate',
-  true: 'checkbox-checked',
-  false: 'checkbox-unchecked',
-};
-const CheckboxSvg = ({
-  checked,
-  disabled,
-}: {
-  checked: boolean | 'indeterminate';
-  disabled?: boolean;
-}) => {
-  const iconClassName = clsx(
-    styles['checkbox__icon'],
-    disabled && styles['checkbox__icon--disabled'],
-  );
-  return (
-    <Icon
-      className={iconClassName}
-      name={
-        checkboxIconNameMap[`${checked}`] as
-          | 'checkbox-indeterminate'
-          | 'checkbox-checked'
-          | 'checkbox-unchecked'
-      }
-      purpose="decorative"
-      size="1.5rem"
-    />
-  );
-};
-
 /**
  * Checkbox input element, exported for greater flexibility.
  * You must provide an `id` prop and connect it to a visible label.
  */
 export const CheckboxInput = React.forwardRef<
   HTMLInputElement,
   CheckboxInputProps
->(({ checked = false, className, disabled, ...other }, ref) => {
-  // Make indeterminate checkbox visually match the colors of a
-  // checked state, but announce itself as "mixed" to screen readers
+>(({ checked, className, disabled, ...other }, ref) => {
+  // Make indeterminate checkbox visually match the colors of a checked state, but announce itself
+  // as "mixed" to screen readers
+  //
+  // However, https://html.spec.whatwg.org/multipage/input.html#checkbox-state-(type=checkbox)
+  // seems to consider `indeterminate` as a separate bit of information than `checked`. They can be
+  // in all 4 combinations (only 3 visual states, though).
+  //
+  // Should we make them separate props?
   const checkedProps =
```

Comment:
Currently we consider indeterminate checkboxes as checked. That's not right, according to https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes.

A prev iteration of this PR fixed that. I reverted it, though. Figured it makes more sense to revisit this in another effort/PR.